octodns/provider/azuredns.py

```diff
@@ -488,6 +488,19 @@ def _apply(self, plan):
         azure_zone_name = desired.name[:len(desired.name) - 1]
         self._check_zone(azure_zone_name, create=True)
 
+        '''
+        Force the operation order to be Delete() before all other operations.
+        Helps avoid problems in updating
+            - a CNAME record into an A record.
+            - an A record into a CNAME record.
+        '''
+
+        for change in changes:
+            class_name = change.__class__.__name__
+            if class_name == 'Delete':
+                getattr(self, '_apply_{}'.format(class_name))(change)
```

Comment:
No need to dynamically look up the apply method since there's an `if` above looking for a specific one.

```suggestion
                self._apply_Delete(change)
```

Reply:
That makes sense, thanks, made the update!